python/cudf/cudf/io/parquet.py

```diff
@@ -824,6 +960,37 @@ def read_parquet(
     # Normalize and validate filters
     filters = _normalize_filters(filters)
 
+    if (
+        engine == "cudf"
+        and filters is not None
+        and categorical_partitions is None
+        and dataset_kwargs is None
+    ):
+        try:
+            ast_filter = translate_filters_to_ast(filters)
+
+            filepaths_or_buffers = ioutils.get_reader_filepath_or_buffer(
+                path_or_data=filepath_or_buffer,
+                fs=fs,
+                storage_options=storage_options,
+                bytes_per_thread=bytes_per_thread,
+            )
+
+            return _parquet_to_frame(
+                filepaths_or_buffers,
+                engine=engine,
+                columns=columns,
+                use_pandas_metadata=use_pandas_metadata,
+                row_groups=None,
+                filters=ast_filter,
+                allow_mismatched_pq_schemas=allow_mismatched_pq_schemas,
+                nrows=nrows,
+                skip_rows=skip_rows,
+            )
+        except NotImplementedError:
+            # Fall back to the pyarrow
+            pass
```

Comment:
Just confirming, we're potentially just expecting `NotImplementedError` can be raised from `translate_filters_to_ast`? If so, can we scope the `try/except` around that call and fallback if that call raises the `NotImplementedError`